src/map.cpp

```diff
@@ -863,9 +863,9 @@ int_fast32_t AStarNodes::getTileWalkCost(const Creature& creature, const Tile* t
 // Floor
 Floor::~Floor()
 {
-	for (uint32_t i = 0; i < FLOOR_SIZE; ++i) {
-		for (uint32_t j = 0; j < FLOOR_SIZE; ++j) {
-			delete tiles[i][j];
+	for (auto& tile : tiles) {
+		for (auto& j : tile) {
```

Comment:
Hmm yeah, I just realized each value is a Tile**, not a Tile*, and it would be more like this:

```
for (Tile** ptr = &tiles[0][0]; ptr < (&tiles[FLOOR_SIZE][FLOOR_SIZE]) + 1; ++ptr) {
  delete *ptr;
}
```

But the range for is much cleaner. I'll adjust it accordingly.